File: backend/app/Models/Venue.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;
use Illuminate\Support\Str;

class Venue extends Model
{
    use SoftDeletes;

    protected $fillable = [
        'user_id', 'name_en', 'name_ar', 'slug', 'description_en', 'description_ar',
        'city', 'district', 'address_en', 'address_ar',
        'latitude', 'longitude', 'capacity', 'cover_image',
        'gallery', 'amenities', 'email', 'phone',
        'verified', 'verification_date', 'status',
    ];

    protected function casts(): array
    {
        return [
            'latitude' => 'decimal:8',
            'longitude' => 'decimal:8',
            'gallery' => 'json',
            'amenities' => 'json',
            'verified' => 'boolean',
            'verification_date' => 'datetime',
        ];
    }

    protected static function booted(): void
    {
        static::creating(function (Venue $venue) {
            if (empty($venue->slug)) {
                $venue->slug = Str::slug($venue->name_en) . '-' . Str::random(6);
            }
        });
    }

    public function user(): \Illuminate\Database\Eloquent\Relations\BelongsTo
    {
        return $this->belongsTo(User::class);
    }

    public function events(): \Illuminate\Database\Eloquent\Relations\HasMany
    {
        return $this->hasMany(Event::class);
    }

    public function scopeByCity($query, string $city)
    {
        return $query->where('city', $city);
    }

    public function scopeActive($query)
    {
        return $query->where('status', 'active');
    }
}

File: mobile/lib/shared/models/venue.dart
class Venue {
  final int id;
  final String nameEn;
  final String? nameAr;
  final String slug;
  final String city;
  final String? district;
  final int? capacity;
  final String? coverImage;
  final List<String>? amenities;

  Venue({
    required this.id,
    required this.nameEn,
    this.nameAr,
    required this.slug,
    required this.city,
    this.district,
    this.capacity,
    this.coverImage,
    this.amenities,
  });

  factory Venue.fromJson(Map<String, dynamic> json) {
    return Venue(
      id: json['id'] as int,
      nameEn: json['name_en'] as String,
      nameAr: json['name_ar'] as String?,
      slug: json['slug'] as String,
      city: json['city'] as String,
      district: json['district'] as String?,
      capacity: json['capacity'] as int?,
      coverImage: json['cover_image'] as String?,
      amenities: (json['amenities'] as List<dynamic>?)
          ?.map((a) => a as String)
          .toList(),
    );
  }

  String get displayName => nameAr ?? nameEn;
}
